hack/get-build-ld-flags.sh

```diff
@@ -37,7 +37,7 @@ fi
 # as a dirty work tree.
 # Additionally, it filters out changes to the `VERSION` file, as this is currently the only way to inject the
 # version-to-build in our pipelines (see https://github.com/gardener/cc-utils/issues/431).
-TREE_STATE="$([ -z "$(git status --porcelain 2>/dev/null | grep -vf <(git ls-files --deleted --ignored --exclude-from=.dockerignore) -e 'VERSION')" ] && echo clean || echo dirty)"
+TREE_STATE="$([ -z "$(git status --porcelain 2>/dev/null | grep -vf <(git ls-files -o --deleted --ignored --exclude-from=.dockerignore) -e 'VERSION')" ] && echo clean || echo dirty)"
```

Comment:
Thanks for looking into this one. I recall that I was quite confused about the change in `ls-files` and I actually had hard times to understand what this piece of code does.
Can you explain what is the breaking change in `ls-files` and how this PR fixes it?

/invite @timebertt

Reply:
Well, this command just filters out changes to files, that are not copied to build containers by `.dockerignore`. 
And with @vpnachev's change it still seems to work for the image builds in the pipeline:
```
$ docker run eu.gcr.io/gardener-project/gardener/apiserver:v1.26.0-dev-0f550f8bdfa28ca44b246ffacfb05e20fc6639e7 /gardener-apiserver --version=raw
version.Info{Major:"1", Minor:"26+", GitVersion:"v1.26.0-dev-0f550f8bdfa28ca44b246ffacfb05e20fc6639e7", GitCommit:"0f550f8bdfa28ca44b246ffacfb05e20fc6639e7", GitTreeState:"clean", BuildDate:"2021-06-29T16:11:36+00:00", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
```

Reply:
Well, ls-files is used to list the files tracked in the repo and we just want to see whether there are some changes in the tracked files and add different suffix to the version. As you have noticed, there is a braking change in the git client introduced with https://github.com/git/git/commit/b338e9f668737e08201c990450b8c3d744f63162.